packages/datagateway-download/src/downloadCart/downloadCartTable.component.test.tsx

```diff
@@ -245,6 +245,37 @@ describe('Download cart table component', () => {
     expect(wrapper.exists('[data-testid="no-selections-message"]')).toBe(true);
   });
 
+  it('disables remove all button while request is processing', async () => {
+    (removeAllDownloadCartItems as jest.Mock).mockImplementation(() => {
+      return new Promise((resolve) => setTimeout(resolve, 2000));
+    });
+
+    const wrapper = createWrapper();
+
+    await act(async () => {
+      await flushPromises();
+      wrapper.update();
+      await flushPromises();
+      wrapper.update();
+    });
+
+    await act(async () => {
+      wrapper.find('button#removeAllButton').simulate('click');
+      await flushPromises();
+      wrapper.update();
+    });
+    expect(
+      wrapper.find('button#removeAllButton').prop('disabled')
+    ).toBeTruthy();
+
+    setTimeout(() => {
+      wrapper.update();
+      expect(wrapper.exists('[data-testid="no-selections-message"]')).toBe(
+        true
+      );
+    }, 2001);
```

Comment:
This is what I'm not a fan of. It manually adds a non-mocked timeout to the test to force the wrapper to update and display the correct thing on the DOM. We want to use fake timers to adhere to good testing practice. Any help on this one would be appreciated!

Reply:
Fake timers and promises inherently don't work well with one another - see https://github.com/ral-facilities/scigateway/blob/169b596d761fc7426136faca946db692849f3f19/src/App.test.tsx#L32 for an example of me working around it. See this issue for info: https://github.com/facebook/jest/issues/10221. One option is to try using the legacy Jest fake timers, as for example there are a few tests in SG which work with old timers and not with the new ones (like the one I linked). However, the only way I could get things working was to manually wait for real timers. Perhaps using a dedicated `sleep` function or even just a `sleep` line of code like `await new Promise((r) => setTimeout(r, 2000));` is more semantic, so you use that line then you wrapper.update/assert to your heart's content rather than within the timeout itself?

Reply:
Thanks for that, I'll use that example as a reference for another attempt. I had a go with the legacy timers, to no avail. Perhaps this will help. I'll ping you again when I have something ready for review :)